fedimint-server/src/consensus/engine.rs

```diff
@@ -472,6 +484,89 @@ impl ConsensusEngine {
             .expect("This is the only place where we write to this key");
     }
 
+    /// Creates the directory within the data directory for storing the database
+    /// checkpoints or deletes checkpoints before `current_session` -
+    /// `checkpoint_retention`.
+    fn initialize_checkpoint_directory(&self, current_session: u64) -> anyhow::Result<()> {
+        let checkpoint_dir = self.data_dir.join("db_checkpoints");
+
+        if checkpoint_dir.exists() {
+            debug!(
+                ?current_session,
+                "Removing database checkpoints up to `current_session`"
+            );
+
+            for checkpoint in fs::read_dir(checkpoint_dir)?.flatten() {
+                // Validate that the directory is a session index
+                if let Ok(file_name) = checkpoint.file_name().into_string() {
+                    if let Ok(session) = file_name.parse::<u64>() {
+                        if session < current_session - self.checkpoint_retention {
```

Comment:
Possibility of  underflow in debug node?

Reply:
Good catch, added check